datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -2752,11 +2752,10 @@ fn calc_func_dependencies_for_project(
         .iter()
         .filter_map(|expr| {
             let expr_name = match expr {
-                Expr::Alias(alias) => {
-                    format!("{}", alias.expr)
-                }
-                _ => format!("{}", expr),
-            };
+                Expr::Alias(alias) => alias.expr.display_name(),
+                _ => expr.display_name(),
+            }
+            .ok()?;
```

Comment:
I don't know whether this makes sense, but we need a name for the function dependencies, so I chose a name that wouldn't be affected by Display.

By the way, I'm confused about why we have so many different names or similar display methods for Expr. Maybe we should organize them or name them more clearly

Reply:
#11782 filed